build.rs

```diff
@@ -1,27 +1,41 @@
+extern crate bindgen;
 extern crate cmake;
 
 fn main() {
-    let mut cfg = cmake::Config::new("openvr");
+    let mut config = cmake::Config::new("openvr");
 
-    // Work around broken cmake build
-    #[cfg(windows)]
-    cfg.cxxflag("/DWIN32");
+    if cfg!(target_os = "macos") {
+        config.define("BUILD_UNIVERSAL", "OFF");
+    } else if cfg!(target_os = "windows") {
+        config.cxxflag("/DWIN32");
+    }
 
-    #[cfg(target_os="macos")]
-    cfg.define("BUILD_UNIVERSAL", "OFF");
-
-    let dst = cfg.build();
+    let dst = config.build();
     println!("cargo:rustc-link-search=native={}/lib", dst.display());
 
-    #[cfg(all(windows, target_pointer_width = "64"))]
-    println!("cargo:rustc-link-lib=static=openvr_api64");
-
-    #[cfg(not(all(windows, target_pointer_width = "64")))]
-    println!("cargo:rustc-link-lib=static=openvr_api");
-
-    #[cfg(target_os="linux")]
-    println!("cargo:rustc-link-lib=stdc++");
-
-    #[cfg(target_os="macos")]
-    println!("cargo:rustc-link-lib=c++");
+    if cfg!(target_os = "linux") {
+        println!("cargo:rustc-link-lib=stdc++");
+        println!("cargo:rustc-link-lib=static=openvr_api");
+    } else if cfg!(target_os = "macos") {
+        println!("cargo:rustc-link-lib=c++");
+        println!("cargo:rustc-link-lib=static=openvr_api");
+    } else if cfg!(target_os = "windows") {
+        println!("cargo:rustc-link-lib=shell32");
+        if cfg!(target_pointer_width = "64") {
+            println!("cargo:rustc-link-lib=static=openvr_api64");
+        } else {
+            println!("cargo:rustc-link-lib=static=openvr_api");
+        }
+    }
+
+    // Generate bindings since enums are i32 on windows and u32 on unix systems
+    // (see https://github.com/rust-lang/rust-bindgen/issues/1244)
+    bindgen::builder()
+        .header("wrapper.hpp")
+        .constified_enum(".*")
+        .prepend_enum_name(false)
+        .generate()
+        .expect("could not generate bindings")
+        .write_to_file("bindings.rs")
+        .expect("could not write bindings.rs");
```

Comment:
Requiring bindgen to be present at build time should be avoided because bindgen depends on LLVM, which is very expensive to build.